include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated.hpp

```diff
@@ -0,0 +1,189 @@
+
+        Propagated::AtmosphericPerturbationType getAtmosphericPerturbationType          ( ) const ;
+
+        Propagated::ThirdBodyPerturbationType getThirdBodyPerturbationType  ( ) const ;
+
+        virtual State           calculateStateAt                            (   const   Instant&                    anInstant                                   ) const override ;
+
+        virtual Integer         calculateRevolutionNumberAt                 (   const   Instant&                    anInstant                                   ) const override ; // [TBR] ?
+        
+        virtual void            print                                       (           std::ostream&               anOutputStream,
+                                                                                        bool                        displayDecorator                            =   true ) const override ;
+
+        static String           StringFromGravitationalPerturbationType     (   const   Propagated::GravitationalPerturbationType&       aGravitationalPerturbationType           ) ;
+
+        static String           StringFromAtmosphericPerturbationType       (   const   Propagated::AtmosphericPerturbationType&      aAtmosphericPerturbationType          ) ;
+
+        static String           StringFromThirdBodyPerturbationType         (   const   Propagated::ThirdBodyPerturbationType&  aThirdBodyPerturbationType      ) ;
+
+    protected:
+
+        virtual bool            operator ==                                 (   const   trajectory::Model&          aModel                                      ) const override ;
+
+        virtual bool            operator !=                                 (   const   trajectory::Model&          aModel                                      ) const override ;
+
+    private:
+
+        State                   state_ ;
+        Instant                 epoch_ ;
+        Derived                 gravitationalParameter_ ;
+        Length                  equatorialRadius_ ;
+        Propagated::GravitationalPerturbationType gravitationalPerturbationType_ ;
+        Propagated::AtmosphericPerturbationType atmosphericPerturbationType_ ;
+        Propagated::ThirdBodyPerturbationType thirdBodyPerturbationType_ ;
+        bool propagationLogStatus_ ;
+
+        static void             TwoBodyDynamics                             (   const   state_type&                 x, 
+                                                                                        state_type&                 dxdt, 
+                                                                                const   double                      t,
+                                                                                const   double&                     mu_SI                                       ) ;
+
+        static void             PropObserver                                (   const   state_type&                 x, 
```

Comment:
I really don't think it needs to be its own class since its so simple... In an ideal world maybe but that sounds like something we shouldn't worry about in the short term